community/spatial-index/src/main/java/org/neo4j/gis/spatial/index/curves/SpaceFillingCurve.java

```diff
@@ -262,8 +262,27 @@ private Long derivedValueFor( long[] normalizedValues, int level )
 
     public List<LongRange> getTilesIntersectingEnvelope( double[] fromOrNull, double[] toOrNull, SpaceFillingCurveConfiguration config )
     {
-        double[] from = fromOrNull == null ? range.getMin() : fromOrNull;
-        double[] to = toOrNull == null ? range.getMax() : toOrNull;
+        double[] from = fromOrNull == null ? range.getMin() : fromOrNull.clone();
+        double[] to = toOrNull == null ? range.getMax() : toOrNull.clone();
```

Comment:
I don't understand why the `clone` is necessary?

Reply:
We don't want to change the limit in the original `GeometryRangePredicate`, without cloning the subsequent filtering will be wrong.

Reply:
OK, got it.